pallets/manta-pay/src/types.rs

```diff
@@ -184,7 +257,171 @@ impl TryFrom<SenderPost> for config::SenderPost {
     fn try_from(post: SenderPost) -> Result<Self, Self::Error> {
         Ok(Self {
             utxo_accumulator_output: decode(post.utxo_accumulator_output)?,
-            void_number: decode(post.void_number)?,
+            nullifier: config::Nullifier {
+                commitment: decode(post.nullifier_commitment)?,
+                outgoing_note: TryFrom::try_from(post.outgoing_note)?,
+            },
+        })
+    }
+}
+
+/// Incoming Ciphertext Type
+pub type IncomingCiphertext = [[u8; 32]; 3];
```

Comment:
Why 3 256-bit blobs `IncomingCiphertext` and 2 256-bit blobs `OutgoingCiphertext`. 

Probably just don't understand the underlying crypto, but am curious

Reply:
Added comments explaining this